incubator/hnc/pkg/reconcilers/hierarchy_config.go

```diff
@@ -188,6 +188,8 @@ func (r *HierarchyConfigReconciler) onMissingNamespace(log logr.Logger, ns *fore
 		if ns.RequiredChildOf != "" {
 			log.Info("Will create missing namespace", "forParent", ns.RequiredChildOf)
 			nsInst.Name = ns.Name()
+			// Set "api.AnnotationOwner" annotation to the non-existent yet namespace.
```

Comment:
As discussed offline, need to skip the current code processing hc.Spec.RequiredChildren

Reply:
I made changes to `syncRequiredChildOf()` and `syncChildren()` functions using the `enable-hns-reconciler` flag.  For the `hns-reconciler-enabled` case, I added TODOs and excluded three hierarchyConfig test cases where the 'RequiredChildConflict' condition is set for conflicts with `spec.requiredChildren` field.